stablehlo/dialect/StablehloOps.cpp

```diff
@@ -2679,19 +2649,16 @@ static ParseResult parseStruct(
     for (const auto& it : llvm::enumerate(keywords)) {
       size_t index = it.index();
       StringRef keyword = it.value();
-      if (succeeded(parser.parseOptionalKeyword(keyword))) {
-        if (seen[index]) {
-          return parser.emitError(parser.getCurrentLocation())
-                 << "duplicated `" << keyword << "` entry";
-        }
-        if (parseEqual.empty() || parseEqual[index]) {
-          if (failed(parser.parseEqual())) return failure();
-        }
-        if (failed(parseFuncs[index]())) return failure();
-        if (failed(parser.parseOptionalComma())) return parser.parseGreater();
-        seen[index] = true;
-        foundOne = true;
-      }
+      if (failed(parser.parseOptionalKeyword(keyword))) continue;
+      if (seen[index])
+        return parser.emitError(parser.getCurrentLocation())
+               << "duplicated `" << keyword << "` entry";
+      if (parseEqual.empty() || parseEqual[index])
+        if (failed(parser.parseEqual())) return failure();
```

Comment:
```
// Use braces on the outer `if` to avoid a potential dangling `else`
// situation.
if (isa<VarDecl>(D)) {
  if (shouldProcessAttr(A))
    handleAttr(A);
}
```